executor/ddl.go

```diff
@@ -635,6 +639,40 @@ func (e *DDLExec) executeRecoverTable(s *ast.RecoverTableStmt) error {
 	return err
 }
 
+// recoverTablePlacement is used when recover/flashback table.
+// It will replace the placement policy of table with the direct options because the original policy may be deleted
+func recoverTablePlacement(snapshotMeta *meta.Meta, tblInfo *model.TableInfo) (*model.TableInfo, error) {
+	if ref := tblInfo.PlacementPolicyRef; ref != nil {
+		policy, err := snapshotMeta.GetPolicy(ref.ID)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
```

Comment:
I don't understand. If the policy is dropped, this function returns an error and the operation fails. How does it solve your problem?

Reply:
We cannot drop a policy which is used by other tables. So when the table is dropped or truncated, the policy must exists. We are using a snapshot meta here with the version where the table has not dropped yet, so we can still get the  policy even if it is dropped now

Reply:
You can check the test cases. I've added some tests there.